### ISSUE TITLE
System analysis using BPMN

### ISSUE THREAD

**Issue by UITS77** [UITS77/help-desk#17]
2
down vote
favorite
I am currently analysing a system of aircraft maintenance company using BPMN.

I have 3 questions regarding BPMN:

First:

enter image description here

Here when the technician orders a new part, he has to wait until either the part is recieved or not. If is recieved then replace the part or stop the work. My question is that is the gateway (event-based) that I use is right or not?

Second:

enter image description here

Here we are talking about the tools if they are available or not. If the tools available then a technician can take them and start the work. If they are not available then they will check if the failure (snag) is small and will not take a long time to be fixed then in this case they will take the tools that are being used in another aircraft (sending request to the technician who is working on that aircraft). If the snag is not small then he has to wait until the tools are available. The problem still with using the proper gateways. Are the gateways that I am using right or not?

enter image description here

Here the problem with aircraft parts. If the supply department has not the part then they contact the job control department that will check if there is any aircraft under maintenance so that they take its part until receiving a new part from the industry. Is my modeling right or not?

So basically all what I want is checking if the modeling that I have done with these parts is right or they need to be changed.

Thanks

**Comment by UITS77** [comment]
Some general information:

BPMN 2.0 is an OMG Standard, so you can always check the specification, in case you are not sure if your way of modeling is correct: http://www.omg.org/spec/BPMN/2.0/
There are tools providing an automatic syntax check, for example Signavio. You can check if the tool you are using has a similar feature.
To your specific questions:

The use of the gateway is correct. However, I would say that "not receiving the new part" might not be a message. Perhaps it is better to use an intermediate timer event here, meaning "if part is not received after time X, stop the work."
Here you definitely don't need to use the event-based Gateways. See BPMN Spec, page 287 (317 in PDF file): > "Gateways are used to control how Sequence Flows interact as they converge and diverge within a Process."
Basically the same as 2.: No event-based gateways necessary. Furthermore, I'd check the syntax here and try to reduce complexity, as the combination of process links and parallel gateways looks rather tricky.

**Comment by UITS77** [comment]
@UITS77 I think there is a problem with setting the right time zone
